devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -489,6 +498,7 @@ private void restoreTerminalState() {
     private void handleAutoCompile() throws MojoExecutionException {
         //we check to see if there was a compile (or later) goal before this plugin
         boolean compileNeeded = true;
+        boolean indexClassNeeded = true;
```

Comment:
I still think the initial value here should depend on whether the `META-INF/jandex.idx` exists in the classes dir. Because if the jandex plugin was configured to run in the `compile` phase (right after the classes have been compiled), with this change the jandex will be called twice.

Reply:
Good point. But why not looking up the phase the execution is bound to?
In `isGoalConfigured()` we already look up the `org.apache.maven.model.PluginExecution` (which has the phase).

Reply:
That could work too.

Reply:
looking at `isGoalConfigured()`, would be better in multi-module project no?

Reply:
I updated the code. I first load the phase which was configured to execute the `jandex` goal. And if that phase is present, then we don't run the `jandex` goal